src/resolver.cpp

```diff
@@ -235,211 +222,95 @@ std::string UsdOpenAssetIOResolver::_CreateIdentifier(
           // (for now).
           // TODO(DF): Allow the manager to provide an identifier representing
-          //  an "anchored" entity reference.
+          //  an "anchored" entity reference via getWithRelationship.
           identifier = assetPath;
         } else {
-          identifier = ArDefaultResolver::_CreateIdentifier(
-              assetPath,
-              locationInManagerContextForEntity(manager_, readContext_, anchorAssetPath));
+          identifier = ArDefaultResolver::_CreateIdentifier(assetPath, anchorAssetPath);
         }
 
         return identifier;
       },
-      logger_, fnName);
-
-  {
-    std::ostringstream logMsg;
-    logMsg << "[tid=" << std::this_thread::get_id() << "] "
-           << "result: " << result;
-    logger_->debug(logMsg.str());
-  }
-  return result;
-}
-
-// TODO(DF): Implement for publishing workflow.
-std::string UsdOpenAssetIOResolver::_CreateIdentifierForNewAsset(
-    const std::string &assetPath, const ArResolvedPath &anchorAssetPath) const {
-  auto result = ArDefaultResolver::_CreateIdentifierForNewAsset(assetPath, anchorAssetPath);
-  logger_->debug(TF_FUNC_NAME());
-
-  logger_->debug("  assetPath: " + assetPath);
-  logger_->debug("  anchorAssetPath: " + anchorAssetPath.GetPathString());
-  logger_->debug("  result: " + result);
-
-  return result;
+      logger_, TF_FUNC_NAME());
 }
 
 ArResolvedPath UsdOpenAssetIOResolver::_Resolve(const std::string &assetPath) const {
-  const std::string fnName = TF_FUNC_NAME();
-  {
-    std::ostringstream logMsg;
-    logMsg << "[tid=" << std::this_thread::get_id() << "] " << fnName
-           << "\n  assetPath: " << assetPath;
-    logger_->debug(logMsg.str());
-  }
-
-  auto result = catchAndLogExceptions(
+  return catchAndLogExceptions(
       [&] {
-        if (manager_->isEntityReferenceString(assetPath)) {
-          // TODO(DF): We may wish to do more here, e.g.
-          //  `finalizedEntityVersion()`
-          return ArResolvedPath{assetPath};
+        if (auto entityReference = manager_->createEntityReferenceIfValid(assetPath)) {
+          return ArResolvedPath{resolveToPath(manager_, readContext_, entityReference.value())};
         }
         return ArDefaultResolver::_Resolve(assetPath);
       },
-      logger_, fnName);
+      logger_, TF_FUNC_NAME());
+}
 
-  {
-    std::ostringstream logMsg;
-    logMsg << "[tid=" << std::this_thread::get_id() << "] "
-           << "result: " << result.GetPathString();
-    logger_->debug(logMsg.str());
-  }
+/*
+ * Write
+ *
+ * We don't currently support writes to OpenAssetIO entity references.
+ * In order to call register when the ArAsset is closed, we'll need
+ * not resolve in _ResolveForNewAsset, and pass the entity reference
+ * through.
+ */
 
-  return result;
+std::string UsdOpenAssetIOResolver::_CreateIdentifierForNewAsset(
+    const std::string &assetPath, const ArResolvedPath &anchorAssetPath) const {
+  if (manager_->isEntityReferenceString(assetPath)) {
```

Comment:
[fix] `anchorAssetPath` may be an entity reference too/instead.

Reply:
I'm not sure I get you. 

Though I do notice now that `anchorAssetPath` is an `ArResolvedPath`, so maybe all good. 

The problem I thought we might have is if `assetPath` isn't an entity reference, but `anchorAssetPath` is, then we'd still fall through to the `ArDefaultResolver` implementation, which won't understand `anchorAssetPath`.

But like I say, if `anchorAssetPath` is always pre-`resolve`d then it's all good.

Reply:
Oh I see what you mean. Yeah - Ill double check what values come in.

Reply:
I checked and `_CreateeIdentifier` always gets the resolved version as the anchor path